src/error.js

```diff
@@ -0,0 +1,15 @@
+import defaultError from './internal/error-handler';
+
+export default (errorHandler) => (app = {}) => {
+  const {error = defaultError} = app;
+  return {
+    ...app,
+    error: (err, req, res) => {
+      try {
+        return errorHandler(err, req, res);
+      } catch (err) {
+        return error(err, req, res);
```

Comment:
We have do decide between calling the upstream handler with the original or caught error. What is more valuable? The original error or the new one? 😕